### PR TITLE
Parse Hex

### PR DESCRIPTION
**Pull request by Jason2605**
# Parse Hex

### What's Changed:

Ability to parse hex codes

#

### Type of Change:

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#

### Housekeeping:

- [x] Tests have been updated to reflect the changes done within this PR (if applicable).
- [x] Documentation has been updated to reflect the changes done within this PR (if applicable).

#

### CODE REVIEW

File: src/vm/compiler.c [hunk 1]
```diff
@@ -1037,6 +1037,28 @@ int parseEscapeSequences(char *string, int length) {
                     string[i + 1] = '\v';
                     break;
                 }
+                case 'x': {
+                    if (string[i + 2] == '\0' || string[i + 3] == '\0') {
+                        error(parser, "\\x escape code expects format \\xhh");
+                        break;
+                    }
+
```
Comment (briandowns) [inline src/vm/compiler.c:1045]: Can we add an additional check here to make sure the given characters are in fact valid hex (0123456789abcdefABCDEF)?

Comment (Jason2605) [reply src/vm/compiler.c:1045]: That should happen below in the strtol check, but potentially an easier check is checking for the char being in that range

Comment (briandowns) [reply src/vm/compiler.c:1045]: Oh my bad, I just noticed that now. I'll defer to you. :)

Comment (Jason2605) [reply src/vm/compiler.c:1045]: No worries! Thanks for double checking anyways

**Pull request merged**